deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -288,22 +296,24 @@ private void ingestFileObjectAsWork(ContentContainerObject parent, Resource pare
 		}
 		Resource workResc = workModel.createResource(workPid.getRepositoryPath());
 		workResc.addProperty(DC.title, label);
+		WorkObject newWork  = null;
 		FedoraTransaction tx = repository.startTransaction();
 		try {
-			WorkObject newWork = repository.createWorkObject(workPid, workModel);
+			newWork = repository.createWorkObject(workPid, workModel);
 
 			addDescription(newWork);
-
-			addFileToWork(newWork, childResc);
+			FileObject fileObj = addFileToWork(newWork, childResc);
+			// add ingestion event for work object
+			addIngestionEventForContainer(newWork, parentResc);
```

Comment:
Mainly for readability and truth in reporting, but could you move this to after `parent.addMember`?